### PR TITLE
Treatment Target

### PR DESCRIPTION
**Pull request by Luke-Sikina**
Treatment Target

# Function
Create two new Study View treatment charts for treatment targets. Treatment targets are the biological function that a drug might target; a good example is `Tyrosine Kinase`. The two charts reflect the sample / patient paradigm already used by the treatment and treatment group charts.

# Implementation
This PR adds the following to the codebase:
- Backend
  - Expanded ClinicalEventKeyCode
    - Delimited value types
    - Added AGENT_TARGET
  - StudyViewFilterApplier
    - Tweeked StudyViewFilterSubApplier hierarchy
    - Added PatientTreatmentTargetSubFilter and SampleTreatmentTargetSubFilter
    - Simplified wiring of StudyViewSubFIlterAppliers
  - Treatment Controller / Service / Repository / Mapper
    - Added support for comma delimited clinical event values
- Frontend
  - API: add treatment target to clinical event key enum
  - StudyViewPageStore:
    - Treatment target data methods
    - Add treatment charts to spaghetti

### CODE REVIEW

File: web/src/main/java/org/cbioportal/web/util/appliers/TreatmentRowExtractor.java [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
 @Service
 public class TreatmentRowExtractor {
     public Set<String> extractSamples(TreatmentRow row) {
-        Set<String> samples = new HashSet<>();
-        row.getSamples().forEach(sample -> {
-            samples.add(sample.key());
-        });
-
-        return samples;
+        return row.getSamples().stream()
```
Comment (Luke-Sikina) [inline web/src/main/java/org/cbioportal/web/util/appliers/TreatmentRowExtractor.java:13]: This logic hasn't changed in practice. The collection is immutable now, but that doesn't matter. No idea why I made this code so gross to begin with.

File: persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/TreatmentMyBatisRepository.java [hunk 2]
```diff
@@ -43,8 +63,17 @@
...
+        return treatmentMapper.getAllUniqueTreatments(sampleIds, studyIds, key.getKey())
+            .stream()
+            .flatMap(treatment -> {
+                if (key.isDelimited()) {
+                    return Arrays.stream(treatment.split(key.getDelimiter()));
+                } else {
+                    return Stream.of(treatment);
+                }
+            })
+            .collect(Collectors.toSet());
```
Comment (dippindots) [inline persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/TreatmentMyBatisRepository.java:76]: I have one question about this. I know `treatmentMapper.getAllUniqueTreatments` can promise give you unique treatments, but how can we know the result is still unique after we do `treatment.split(key.getDelimiter())`?

File: web/src/main/java/org/cbioportal/web/util/appliers/PatientTreatmentTargetFilterApplier.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+public class PatientTreatmentTargetFilterApplier extends AbstractPatientTreatmentFilter {
+
+    @Override
+    protected AndedPatientTreatmentFilters getFilters(StudyViewFilter filter) {
+        return filter.getPatientTreatmentTargetFilters();
+    }
+
+    @Override
+    protected ClinicalEventKeyCode getCode() {
+        return ClinicalEventKeyCode.AgentTarget;
```
Comment (dippindots) [inline web/src/main/java/org/cbioportal/web/util/appliers/PatientTreatmentTargetFilterApplier.java:18]: So taking patient data as an example
`Agent` -> `PatientTreatment`
`AgentClass` -> `PatientTreatmentGroup`
`AgentTarget` -> `PatientTreatmentTarget`
Is this correct?

File: web/src/test/resources/springfox.properties [hunk 4]
```diff
@@ -0,0 +1 @@
+springfox.documentation.swagger.v2.path:/api-docs
```
Comment (dippindots) [inline web/src/test/resources/springfox.properties:1]: What's this for?

**Review by dippindots — COMMENTED** [review]
Thanks @Luke-Sikina! Overall it looks good to me. I have some questions, I put them in the comments. And I see the core tests are failing, I rerun it but still get error, so I think it's related to this pr, could you take a look?

File: persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/TreatmentMyBatisRepository.java (hunk 2 above)
Comment (Luke-Sikina) [reply persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/TreatmentMyBatisRepository.java:76]: We don't, but `toSet` will enforce uniqueness, so you'll still end up with the correct output.

File: web/src/main/java/org/cbioportal/web/util/appliers/PatientTreatmentTargetFilterApplier.java (hunk 3 above)
Comment (Luke-Sikina) [reply web/src/main/java/org/cbioportal/web/util/appliers/PatientTreatmentTargetFilterApplier.java:18]: Yup

File: web/src/test/resources/springfox.properties (hunk 4 above)
Comment (Luke-Sikina) [reply web/src/test/resources/springfox.properties:1]: Oop, think that's a holdover from another PR. Deleted.

File: persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/TreatmentMyBatisRepository.java (hunk 2 above)
Comment (dippindots) [reply persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/TreatmentMyBatisRepository.java:76]: I ignored that, you are right

**Pull request merged**